packages/remix-react/components.tsx

```diff
@@ -308,35 +313,42 @@ let NavLink = React.forwardRef<HTMLAnchorElement, RemixNavLinkProps>(
           {...props}
           {...prefetchHandlers}
         />
-        {shouldPrefetch ? <PrefetchPageLinks page={href} /> : null}
+        {shouldPrefetch && !isAbsolute ? (
+          <PrefetchPageLinks page={href} />
+        ) : null}
       </>
     );
   }
 );
 NavLink.displayName = "NavLink";
 export { NavLink };
+
 /**
  * This component renders an anchor tag and is the primary way the user will
  * navigate around your website.
  *
  * @see https://remix.run/api/remix#link
  */
 let Link = React.forwardRef<HTMLAnchorElement, RemixLinkProps>(
-  ({ to, prefetch = "none", ...props }, forwardedRef) => {
-    let href = useHref(to);
+  ({ to, prefetch = "none", children, ...props }, forwardedRef) => {
+    let location = typeof to === "string" ? to : createPath(to);
+    let isAbsolute =
+      /^[a-z+]+:\/\//i.test(location) || location.startsWith("//");
+
+    let href = useHref(location);
     let [shouldPrefetch, prefetchHandlers] = usePrefetchBehavior(
       prefetch,
       props
     );
+
     return (
       <>
-        <RouterLink
-          ref={forwardedRef}
-          to={to}
-          {...props}
-          {...prefetchHandlers}
-        />
-        {shouldPrefetch ? <PrefetchPageLinks page={href} /> : null}
+        <RouterLink ref={forwardedRef} to={to} {...props} {...prefetchHandlers}>
+          {children}
+        </RouterLink>
```

Comment:
Do we need to pull this out or could it stay in `props` and get spread on?

Reply:
probably fine to keep in props and spread, i.. i just don't like that lmao